File: 3/I/main.go
package main

import "fmt"

func main() {
	var n, m int
	var s string
	l := map[string]int{}

	fmt.Scanf("%d", &n)

	a := make([]string, 0, n)

	for i := 1; i < n; i++ {
		fmt.Scanf("%d", &m)
		for j := 0; j < m; j++ {
			fmt.Scanf("%s", &s)
			l[s]++
		}
	}

	fmt.Scanf("%d", &m)
	for j := 0; j < m; j++ {
		fmt.Scanf("%s", &s)
		l[s]++
		if l[s] == n {
			a = append(a, s)
		}
	}

	fmt.Println(len(a))
	for _, v := range a {
		fmt.Println(v)
	}

	fmt.Println(len(l))
	for k := range l {
		fmt.Println(k)
	}
}

File: 3/I/main.py
n = int(input())
d = dict()
l = []

for i in range(n-1):
    m = int(input())
    for j in range(m):
        s = input()
        if s not in d:
            d[s]=1
        else:
            d[s] += 1

m = int(input())
for j in range(m):
    s = input()
    if s not in d:
        d[s]=1
    elif d[s] == n-1:
        l.append(s)

print(len(l))
for v in l:
    print(v)
print(len(d))
for k in d:
    print(k)